be/src/exprs/array_functions.cpp

```diff
@@ -1199,4 +1206,213 @@ StatusOr<ColumnPtr> ArrayFunctions::array_slice(FunctionContext* ctx, const Colu
+                        is_contains = true;
+                        break;
+                    }
+                }
+
+                if (!is_contains) {
+                    result_elements->append(*elements, elements_idx, 1);
+                }
+            }
+        }
+
+        result_offsets->append(result_elements->size());
+    }
+
+    return NullableColumn::create(ArrayColumn::create(std::move(result_elements), result_offsets), array_null);
+}
+
+StatusOr<ColumnPtr> ArrayFunctions::array_reverse_any_types(FunctionContext* ctx, const Columns& columns) {
+    DCHECK_EQ(1, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto [array_null, elements, offsets] = unpack_array_column(columns[0]);
+    auto* offsets_ptr = offsets->get_data().data();
+
+    auto result_elements = elements->clone_empty();
+
+    size_t rows = columns[0]->size();
+    for (auto i = 0; i < rows; i++) {
+        size_t offset = offsets_ptr[i];
+        int64_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
+
+        for (int64_t j = array_size - 1; j >= 0; j--) {
+            result_elements->append(*elements, offset + j, 1);
+        }
+    }
+
+    return NullableColumn::create(ArrayColumn::create(std::move(result_elements), UInt32Column::create(*offsets)),
+                                  array_null);
+}
+
+inline static void nestloop_intersect(uint8_t* hits, const Column* base, size_t base_start, size_t base_end,
+                                      const Column* cmp, size_t cmp_start, size_t cmp_end) {
+    for (size_t base_ele_idx = base_start; base_ele_idx < base_end; base_ele_idx++) {
+        if (hits[base_ele_idx] == 0) {
+            continue;
+        }
+
+        size_t cmp_idx = cmp_start;
+        for (; cmp_idx < cmp_end; cmp_idx++) {
+            if (base->equals(base_ele_idx, *cmp, cmp_idx)) {
+                break;
+            }
+        }
+
+        if (cmp_idx >= cmp_end) {
+            hits[base_ele_idx] = 0;
+        }
+    }
+}
+
+StatusOr<ColumnPtr> ArrayFunctions::array_intersect_any_type(FunctionContext* ctx, const Columns& columns) {
+    DCHECK_LE(2, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    size_t rows = columns[0]->size();
+    size_t usage = columns[0]->memory_usage();
+    ColumnPtr base_col = columns[0];
+    int base_idx = 0;
+
+    auto nulls = NullColumn::create(rows, 0);
+    // find minimum column
+    for (size_t i = 0; i < columns.size(); i++) {
+        if (columns[i]->memory_usage() < usage) {
+            base_col = columns[i];
+            base_idx = i;
+            usage = columns[i]->memory_usage();
+        }
+
+        FunctionHelper::union_produce_nullable_column(columns[i], &nulls);
+    }
+
+    // do distinct first
+    auto distinct_col = array_distinct_any_type(ctx, {base_col});
+    DCHECK(distinct_col.ok());
+
+    auto* dis_array_col =
+            down_cast<ArrayColumn*>(down_cast<NullableColumn*>(distinct_col.value().get())->data_column().get());
```

Comment:
is it must be nullabe column? `down_cast<NullableColumn*>(distinct_col.value().get())`?

Reply:
directly use ColumnHelper:get_data_colun()